scipy/spatial/transform/rotation.py

```diff
@@ -657,3 +656,106 @@ def __mul__(self, other):
         return self.__class__(result, normalized=True)
+
+
+class Slerp(object):
+    """Spherical Linear Interpolation of Rotations.
+
+    The interpolation is done using the Spherical Linear intERPolation
+    (SLERP) algorithm given in [1]_. This ensures that the interpolated
+    quaternions follow the shortest path between initial and final
+    orientations, and results in rotation around a single axis with
+    constant velocity.
+
+    Given a set of `rotations` and corresponding fixed `timestamps`, this class
+    will initialize a `Slerp` instance which is an interpolator. The
+    interpolator can be called with another set of timestamps to return a
+    single `Rotation` object containing the interpolated rotations.
+
+    Parameters
+    ----------
+    t : array_like, shape (N,)
+        An array of strictly increasing timestamps for the fixed rotations. The
+        number of times `N` must be equal to the number of rotations specified
+        in the `rotations` object.
+    r : a `Rotation` instance
+        This object contains the fixed rotations between which the
+        interpolation will be done. Must contain `N`rotations and at least 2.
+
+    Methods
+    -------
+    __call__
+
+    References
+    ----------
+    .. [1] `Quaternion Slerp
+            <https://en.wikipedia.org/wiki/Slerp#Quaternion_Slerp>`_
+    """
+    def __init__(self, t, r):
+        self._key_quats = r._quat
+        self._num_rots = r._quat.shape[0]
+        self._times = np.asarray(t)
+
+        if self._num_rots == 1:
+            raise ValueError("Expected at least 2 rotations in rotation object"
+                             ", got {}.".format(self._num_rots))
+
+        if self._times.ndim > 1:
+            raise ValueError("Expected keyframe timestamp array to be 1 "
+                             "dimensional, got {} dimensions.".format(
+                                self._times.ndim))
+
+        if self._times.shape[0] != self._num_rots:
+            raise ValueError("Expected number of rotations to be equal to "
+                             "number of keyframe timestamps, got {} rotations "
+                             "and {} timestamps.".format(
+                                self._num_rots,
+                                self._times.shape[0],
+                             ))
+
+        # Make a copy for modification
+        q0 = self._key_quats[0:-1].copy()
+        q0[:, -1] *= -1
+        q1 = self._key_quats[1:]
+        self._interp_rotvecs = Rotation.from_quaternion(
+            _compose_quat(q0, q1)).as_rotvec()
+
+    def __call__(self, timestamps):
+        """Generate interpolated rotations at given times.
+
+        Given an array of `timestamps` returns a `Rotation` instance with the
+        interpolated rotations at the given times.
+
+        Parameters
+        ----------
+        timestamps : array_like, 1D
+            An array of increasing timestamps, for the interpolated rotations
```

Comment:
Don't think they need to be in increasing order. Think of better and shorter phrasing too.